x-pack/elastic-agent/pkg/composable/providers/kubernetes/pod.go

```diff
@@ -217,8 +217,7 @@ func generatePodData(
 	k8sMapping := map[string]interface{}(kubemetaMap.(common.MapStr).Clone())
 
 	if len(namespaceAnnotations) != 0 {
-		// TODO: convert it to namespace.annotations for 8.0
-		k8sMapping["namespace_annotations"] = namespaceAnnotations
+		k8sMapping["namespace"].(common.MapStr).Put("annotations", namespaceAnnotations)
```

Comment:
If `k8sMapping["namespace"]` return `nil` it will panic. Perhaps you can check the value first or you can just add the `namespace.annotations` key at top level at `kubemetaMap.(common.MapStr).Clone()`. Sth like this:
```
ckMeta := kubemetaMap.(common.MapStr).Clone()
if len(namespaceAnnotations) != 0 {
    ckMeta.Put("kubernetes.annotations", namespaceAnnotations)
}
k8sMapping := map[string]interface{}(ckMeta)
```

Reply:
TBH I thought there is no way that this can be nil for a pod or a service

Reply:
I updated it